frontend/src/store/secret.js

```diff
@@ -90,27 +94,25 @@ export const useSecretStore = defineStore('secret', () => {
 
   const infrastructureSecretList = computed(() => {
     return filter(list.value, secret => {
-      return !!secret.metadata.cloudProviderKind
+      return cloudProfileStore.sortedProviderTypesList.includes(secret.metadata.provider?.type)
     })
   })
 
   const dnsSecretList = computed(() => {
     return filter(list.value, secret => {
-      return !!secret.metadata.dnsProviderName && isOwnSecret(secret) // secret binding not supported
+      return gardenerExtensionStore.dnsProviderTypes.includes(secret.metadata.provider?.type) && isOwnSecret(secret) // secret binding not supported
     })
   })
 
   function getCloudProviderSecretByName ({ namespace, name }) {
     return find(list.value, eqlNameAndNamespace({ name, namespace }))
   }
 
-  function infrastructureSecretsByCloudProfileName (cloudProfileName) {
-    return filter(list.value, ['metadata.cloudProfileName', cloudProfileName])
-  }
-
-  function dnsSecretsByProviderKind (dnsProviderName) {
+  function secretsByProviderType (providerType) {
```

Comment:
Use a factory function to create computed properties based on `providerType`

```js
function useSecretsByProviderType(providerType) {
  return computed(() => {
    const isDnsSecret = ..... // rest of the code
  })
}
```

Reply:
we currently don't use factory functions for the other functions why do we want to change it for this one?

Reply:
I created a dedicated composable `useSecretList`